### ISSUE TITLE
Question about the Performances of SegContrast

### ISSUE THREAD

**Issue by auniquesun** [PRBonn/segcontrast#4]
@nuneslu Thanks for sharing the paper and code.

As we know, in recent years, self-supervised learning on 3D point cloud attracts more and more investigation. PointContrast and DepthContrast have been mentioned in the paper. There are also other representative work OcCo (ICCV 2021), PoinTr (ICCV 2021), CrossPoint (CVPR 2022), Point-Bert (CVPR 2022)

According to the results in paper, SegContrast outperforms previous methods, sometimes by a large margin.

I wonder what makes this happen and what the differences are with existing methods, e.g. the design mechanism, the dataset, or preprocessing steps for point clouds, since it seems that the backbone, the contrastive loss and the pre-training technology used in SegContrast are also used in previous work and, the paper gives little section on the ablation studies.

**Comment by nuneslu** [comment]
Hi @auniquesun,

That is true, our method has a similar structure to previous works, but we propose a data-driven method. Instead of extracting a scan-wise feature vector (as in DepthContrast) or sampling random points from the point cloud (PointContrast) to compute the contrastive loss, we extract meaningful structures from the LiDAR point cloud to compute the loss. So even though the goal is still the same, our method maximizes the similarity between similar structures. 

From the linear evaluation (Table V on the paper), one can see that the features learned from the DepthContrast pre-training lack descriptive features to distinguish the different semantic classes, compared to the other methods. PointContrast has a better mIoU but from the class-wise evaluation, one can see that it is mainly because it performs better on the classes that have the most amount of points in the dataset (road, car, building). This happens probably because the method randomly sample points to compute the contrastive loss, and then most of the points will be from these most recurrent classes. Our method is looking toward maximizing the similarity of the structures in the scene, therefore it can learn a more "fine-grained" embedding space, compared to the baselines.
